### PR TITLE
SLICE-115 - Support for List, Set and Collection mapping with @Children

### PR DESCRIPTION
**Pull request by michalwronowski**

### CODE REVIEW

File: slice-test/slice-mapper-test/src/test/groovy/com/cognifide/slice/mapper/annotation/ChildrenTest.groovy [hunk 1]
```diff
@@ -93,6 +93,82 @@ class ChildrenTest extends BaseSetup {
...
+	}
+
+	def "Children test (mapping to Set)"() {
+		setup: "Creating initial content"
+		pageBuilder.content {
+			test("cq:PageContent") {
+				"jcr:content"("text": "Test") {
+					"children"("sling:Folder") {
+						"item"("text": "Test1", "style": "Style1", "size": 1)
+						"item_0"("text": "Test2", "style": "Style2", "size": 2)
```
Comment (mmajchrzak) [inline slice-test/slice-mapper-test/src/test/groovy/com/cognifide/slice/mapper/annotation/ChildrenTest.groovy]: Add more items so we can verify that unique models are stored in resulting Set. Currently, this test doesn't check it.

File: slice-test/slice-mapper-test/src/test/groovy/com/cognifide/slice/mapper/annotation/ChildrenTest.groovy [hunk 2]
```diff
@@ -93,6 +93,82 @@ class ChildrenTest extends BaseSetup {
...
+	}
+
+	def "Children test (mapping to SortedSet)"() {
+		setup: "Creating initial content"
+		pageBuilder.content {
+			test("cq:PageContent") {
+				"jcr:content"("text": "Test") {
+					"children"("sling:Folder") {
+						"item"("text": "Test1", "style": "Style1", "size": 1)
+						"item_0"("text": "Test2", "style": "Style2", "size": 2)
```
Comment (mmajchrzak) [inline slice-test/slice-mapper-test/src/test/groovy/com/cognifide/slice/mapper/annotation/ChildrenTest.groovy]: Add more items and play with order so we can verify that sorted set is returned. Currently, this test doesn't check it.

File: slice-mapper/src/main/java/com/cognifide/slice/mapper/impl/processor/ChildrenFieldProcessor.java [hunk 3]
```diff
@@ -67,9 +72,21 @@ public Object mapResourceToField(Resource resource, ValueMap valueMap, Field fie
...
-		} else {
-			return mappedModels;
 		}
+		return getMappedModelsCollection(mappedModels, fieldType);
+	}
+
+	private Collection getMappedModelsCollection(List<?> mappedModels, Class<?> fieldType) {
+		AssignableFieldTypes assignableFieldType = AssignableFieldTypes.byClass(fieldType);
+		if (assignableFieldType != null) {
+			switch (assignableFieldType) {
```
Comment (mleczey) [inline slice-mapper/src/main/java/com/cognifide/slice/mapper/impl/processor/ChildrenFieldProcessor.java]: What do you think about moving responsibility from this class to AssignableFieldTypes class. For example by changing enum to Map holding types (interfaces) as key and concrete implementation as values?

**Comment by kwielgosz** [comment]
approved

**Pull request merged**